rust-lang/cargo
# perf(package): Speed up verify with 'check'

### What does this PR try to resolve?

In testing workspace publishing, I was really wishing for the build to be faster when I noticed the sea of status messages was "Compiling", instead of "Checking".
I've been glossing over those messages for years and never noticed!

This makes builds faster by skipping the compiler backend / codegen at the cost of not getting post-monomorphization errors. That seems like a small price to pay.

Fixes #14941

### How should we test and review this PR?

I searched through the issues, open and closed, and saw no previous discussions of this.  In particular, I was looking to see if this was previously rejected.
My only assumption is this was missed when `cargo check` was added.

### Additional information
